docs/experimental/bicep-mcp-server.md

```diff
@@ -0,0 +1,46 @@
+# Using Bicep MCP Server in VS Code (Preview!)
+
+> [!NOTE]
+> This feature is currently in Preview while we collect feedback.
+
+## What is it?
+The Bicep VS Code extension now includes a Bicep MCP server with agentic tools to support Bicep code generation for AI agents in VS Code. Model Context Protocol (MCP) is an open standard that enables AI models to interact with external tools and services through a unified interface. 
+
+### What is MCP? 
+MCP follows a client-server architecture:
+- MCP clients (like VS Code) connect to MCP servers and request actions on behalf of the AI model.
+- MCP servers provide one or more tools that expose specific functionalities through a well-defined interface.
+- The Model Context Protocol (MCP) defines the message format for communication between clients and servers, including tool discovery, invocation, and response handling
+
+### Available Bicep MCP tools 
+- `ListAzResourceTypesForProvider`: Lists all available Azure resource types for a specific provider.The return value is a newline-separated list of resource types including their API version, e.g. `Microsoft.KeyVault/vaults@2024-11-01`.Such information is the most accurate and up-to-date as it is sourced from the Azure Resource Provider APIs. 
+- `GetAzResourceTypeSchema`: Gets the schema for a specific Azure resource type and API version. Such information is the most accurate and up-to-date as it is sourced from the Azure Resource Provider APIs. 
+- `GetBicepBestPractices`:  Lists up-to-date recommended Bicep best-practices for authoring templates. These practices help improve maintainability, security, and reliability of your Bicep files. This is helpful additional context if you've been asked to generate Bicep code. 
+
+
+## How to use it?
+### Prerequisites
+- Install the latest version of Visual Studio Code [TODO: LINK]
+- Access to Copilot [TODO: LINK] 
+
+### Using the Bicep tools available tools in Agent Mode 
+
+1. Open the GitHub copilot extension window and select "Agent Mode".
+<img width="985" height="527" alt="image" src="https://github.com/user-attachments/assets/550cab6b-9d8b-450e-9cab-f015bf48afc3" />
+
+2. Click on the tool icon in the GitHub copilot chat window and search for "Bicep (PREVIEW). 
+<img width="697" height="504" alt="image" src="https://github.com/user-attachments/assets/3b68977c-01dc-4eae-b08f-47a795d6564f" />
```

Comment:
This shouldn't actually be necessary to start using it, but I think it's helpful to include information about how to see it. Maybe we could move this to a separate section of the doc?